RHIDP-3024: RHDH 1.2.2 release notes (#380)
redhat-developer/red-hat-developers-documentation-rhdh: modules/release-notes/con-relnotes-fixed-issues.adoc

[id='con-relnotes-fixed-issues_{context}']
= Fixed issues in {product} {product-version} and 1.2.2

== Fixed issues in {product} 1.2.2

This section lists fixed issues with {product} 1.2.2:

Dynamic plugins fail to load if the file name in ConfigMap is invalid::
+
--
Previously, an invalid dynamic plugin file name in ConfigMap prevented the creation of a Backstage CR. With this release, the Backstage CR can be created even with an invalid plugin file name, and a corresponding log message will be displayed if the {product-short} build fails.
--

Plugin ID is missing in the permission table in RBAC Administration UI::
+
--
Previously, plugin IDs were missing from the permission table in UI accessed from *Administration* -> *RBAC*. With this release, updates to both the RBAC backend and frontend plugins now ensure that plugin IDs are displayed in the table.
--

Git information for Helm chart-based workflows is no longer shown in the ArgoCD plugin UI::
+
--

> Previously, the ArgoCD plugin UI displayed Git information for all workflows, including those that were not based on Git. With this release, the ArgoCD plugin UI now only displays Git information for workflows where it is relevant, ensuring that only pertinent information is shown.

Reviewer: Needlessly verbose description of the situation and its resolution by repeating information that's obvious, ensuring that a longer sentence is used to convey the same information. With this verbesge, you are consistent with the style guide and achieve a larger word count. 

+1 to merge, though I prefer the more terse description I provided.

Author: Thank you for the feedback. When not including Jira links, providing context for bug fixes is important. If we add the links, a brief one-liner description of the fix should suffice. We're planning to add Jira links starting with the 1.3 release, so our approach to describing bug fixes may change in future releases.

[id='con-relnotes-fixed-issues_{context}']
= Fixed issues in {product} {product-version} and 1.2.2

== Fixed issues in {product} 1.2.2

This section lists fixed issues with {product} 1.2.2:

Dynamic plugins fail to load if the file name in ConfigMap is invalid::
+
--
Previously, an invalid dynamic plugin file name in ConfigMap prevented the creation of a Backstage CR. With this release, the Backstage CR can be created even with an invalid plugin file name, and a corresponding log message will be displayed if the {product-short} build fails.
--

Plugin ID is missing in the permission table in RBAC Administration UI::
+
--

> Previously, the plugin ID was missing from the permission table in the *Administration* -> *RBAC*. With this release, updates to both the RBAC backend and frontend plugins now ensure the plugin ID is displayed in the table.

Reviewer: I'm assuming this refers to more than one plugin's ID, not just the single entry for the RBAC plugin? Unclear from the way this is written... 

```suggestion
Previously, the plugin ID was missing from the permission table in UI accessed from *Administration* -> *RBAC*. With this release, updates to both the RBAC backend and frontend plugins now ensure that plugin IDs are displayed in the table.
```

Author: Done

--

Deprecation warning about `backend.auth.keys` in application log::
+
--
Previously, {product-short} build logs showed the following deprecation warning:

`DEPRECATION WARNING: The backend.auth.keys config has been replaced by backend.auth.externalAccess. See https://backstage.io/docs/auth/service-to-service-auth`

> With this release, if your deployment uses the default configuration, the deprecation warning will no longer appear in the application logs.

Reviewer: because we updated to the latest? Because we're hiding the warning? I feel like some context is missing here.

Reviewer: can you point at the code where this change occurred? JIRA?

Author: Done, Moved this to known issues.

--

> Commit section in ArgoCD plugin UI::

Author: Thank you for bringing up the point about linking to Jira. We hear you and are planning to include links to bug fixes in version 1.3. However, since we didn't add the links in version 1.2, we're continuing with the same approach for version 1.2.2.

> Commit section in ArgoCD plugin UI::

Reviewer: ```suggestion
Git info for helm chart based workflows no longer shown in ArgoCD plugin UI::
```

Author: Changed the content as follows:

Git information for Helm chart-based workflows is no longer shown in the ArgoCD plugin UI::
+
--
Previously, the ArgoCD plugin UI displayed Git information for all workflows, including those that were not based on Git. With this release, the ArgoCD plugin UI now only displays Git information for workflows where it is relevant, ensuring that only pertinent information is shown.
--

+
--

> Previously, the Git commit section continued to load when viewing the ArgoCD plugin. With this release, the Git commit section in the environment card will no longer appear for ArgoCD plugin UI in the applications deployed with Helm Chart through the ArgoCD UI or CLI.

Reviewer: ```suggestion
Previously, the Git commit section continued to load when viewing the ArgoCD plugin. With this release, the Git commit section in the environment card will no longer appear for ArgoCD plugin UI.
```

Author: Question: Was the installation method the cause of the bug? If so, we should definitely mention it.

Reviewer: no idea. link to the JIRA?

--

Plugin ID is missing in the permission table in RBAC Administration UI::
+
--
Previously, the plugin ID was missing from the permission table in the *Administration* -> *RBAC*. With this release, updates to both the RBAC backend and frontend plugins now ensure the plugin ID is displayed in the table.
--

Deprecation warning about `backend.auth.keys` in application log::
+
--
Previously, {product-short} build logs showed the following deprecation warning:

`DEPRECATION WARNING: The backend.auth.keys config has been replaced by backend.auth.externalAccess. See https://backstage.io/docs/auth/service-to-service-auth`

With this release, if your deployment uses the default configuration, the deprecation warning will no longer appear in the application logs.
--

Commit section in ArgoCD plugin UI::
+
--
Previously, the Git commit section continued to load when viewing the ArgoCD plugin. With this release, the Git commit section in the environment card will no longer appear for ArgoCD plugin UI in the applications deployed with Helm Chart through the ArgoCD UI or CLI.
--

> == Fixed issues in {product} {product-version}

Reviewer 2: There are unuseful duplicated lines. From https://redhat-developer.github.io/red-hat-developers-documentation-rhdh/pr-380/rel-notes-rhdh/  please check the 'table of content', section 6.1 to 6.4. They look incorrect/redundant.

Author: Done, fixed this issue.

//link:{LinkRHIDPIssue}RHIDP-2931[RHIDP-2931]
--

> == Fixed issues in {product} 1.2.2

Reviewer 3: @hmanwani-rh In the RNs for 1.1, we displayed the fixed issues in reverse chronological order (as was the case in some other products that I reviewed). If we want to maintain this pracrice, this section should come immediately after the main heading see https://docs.redhat.com/en/documentation/red_hat_developer_hub/1.1/html/release_notes_for_red_hat_developer_hub_1.1/con-relnotes-fixed-issues_release-notes-rhdh#enhancements-and-bug-fixes-in-red-hat-developer-hub-1-1-2

This section lists fixed issues with {product} 1.2.2:

Dynamic plugins fail to load if the file name in ConfigMap is invalid::
+
--

> Previously, an invalid dynamic plugin file name in ConfigMap prevented the creation of a Backstage CR. With this release, the Backstage CR can be created even with an invalid plugin file name, and a corresponding log message will be displayed if the RHDH build fails.

Reviewer 3: For consistency...in this section anyway:

```suggestion
Previously, an invalid dynamic plugin file name in ConfigMap prevented the creation of a Backstage CR. With this release, the Backstage CR can be created even with an invalid plugin file name, and a corresponding log message will be displayed if the {product-short} build fails.
```

Reviewer 3: @hmanwani-rh I also noticed an invalid attribute in _1.1. Upgrading the Red Hat Developer Hub Helm Chart_

> Log in to the OpenShift cluster as the cluster administrator and switch to the project or namespace in which {prod-short} was installed.

Do you want to fix it in this release update?

Author: Sure, fixed it

--

Plugin ID is missing in the permission table in RBAC Administration UI::
+
--
Previously, the plugin ID was missing from the permission table in the *Administration* -> *RBAC*. With this release, updates to both the RBAC backend and frontend plugins now ensure the plugin ID is displayed in the table.
--

Deprecation warning about `backend.auth.keys` in application log::
+
--

> Previously, RHDH build logs showed the following deprecation warning:

Reviewer 3: ```suggestion
Previously, the {product-short} build logs showed the following deprecation warning:
```

//link:{LinkRHIDPIssue}RHIDP-2931[RHIDP-2931]
--

> == Fixed security issues

Reviewer 2: Found a cosmetic issue :)

This line should be changed to 
`== Fixed security issues in {product} 1.2.2`
and the next line (`=== Fixed security issues in {product} 1.2.2`) should be removed.

Reviewer 3: @josephca Not necessarily, as we separate the issues by z-stream release and list them in reverse chronological order - see section _6. Fixed issues in Red Hat Developer Hub 1.2 and 1.2.2_

However, section **_6.4. Fixed security issues in Red Hat Developer Hub 1.2_** should be a sub-heading of **_6.3. Fixed security issues_**

Like so:
_6.3. Fixed security issues in Red Hat Developer Hub 1.2 and 1.2.2
6.3.1. Fixed security issues in Red Hat Developer Hub 1.2.2
6.3.2 Fixed security issues in Red Hat Developer Hub 1.2_
